docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -305,6 +305,50 @@ In the example below:
 If you are dynamically switching the `leafField` or `mainGroupingCriteria`, the sorting and filtering models will not be cleaned up automatically, and the sorting/filtering will not be re-applied.
 :::
 
+## Automatic parents and children selection
+
+By default, selecting a parent row does not select its children.
+You can override this behavior by using the `rowSelectionPropagation` prop.
+
+Here's how it's structured:
+
+```ts
+type GridRowSelectionPropagation = {
+  descendants?: boolean; // default: false
+  parents?: boolean; // default: false
+};
+```
+
+When `rowSelectionPropagation.descendants` is set to `true`.
+
+- Selecting a parent would auto-select all its filtered descendants.
+- Deselecting a parent row would auto-deselect all its filtered descendants.
+
+When `rowSelectionPropagation.parents` is set to `true`.
+
+- Selecting all the filtered descendants of a parent would auto-select the parent.
+- Deselecting a descendant of a selected parent would auto-deselect the parent.
+
+The example below demonstrates the usage of the `rowSelectionPropagation` prop.
+
+{{"demo": "RowGroupingPropagateSelection.js", "bg": "inline", "defaultCodeOpen": false}}
+
+:::info
+The row selection propagation also affects the "Select all" checkbox in a similar way like any other group checkbox.
+:::
+
+:::info
+The selected unfiltered rows will be auto-deselected when the filter is applied. Row selection propagation is not applied to the unfiltered rows.
+:::
+
+:::warning
+If `props.disableMultipleRowSelection` is set to `true`, the row selection propagation feature would not work.
+:::
+
+:::warning
+The row selection propagation is a client-side feature and not recommended to be used with the [server-side data source](/x/react-data-grid/server-side-data/), since it would work on the partially loaded data only.
+:::
+
```

Comment:
`indeterminateCheckboxAction` is not mentioned anywhere

Reply:
We have a section in row selection: https://mui.com/x/react-data-grid/row-selection/#customize-indeterminate-checkbox-behavior

Does replicating it here make sense for better discoverability?

IMO, we might avoid it if we wanna remove that prop in v8 as suggested in https://github.com/mui/mui-x/issues/14300#issuecomment-2313923763

Reply:
No, you can leave it as is then